Documentation/Basic-Usage.md

```diff
@@ -128,7 +174,13 @@ extension User {
 }
 ```
 
-You could see how this would get much worse with a more complex model.
+Not only is that code much more verbose than the equivalent code using Argo,
+it's also not as safe. It does check to make sure that `id` and `name` are
+non-nil, but it's not checking to see that their types are correct. If that code
+encountered JSON where the `"id"` key returned, say a `String` instead of an
+`Int`, this would lead to a crash.
```

Comment:
Also, maybe not for this doc, but this _only_ works for std lib types. You can't just cast to `MyCoolRelationshipType` and have it work.